### PR TITLE
Get "mr test" running on openqaworker15 for 15 SP4

### PR DESCRIPTION
**Pull request by lilyeyes**
Get "mr test" running on openqaworker15 for 15 SP4

jsc#TEAM-6726 - Get "mr test" running on openqaworker15 for 15 SP4 (intel on-premise)
(https://jira.suse.com/browse/TEAM-6726)

NOTE:
1. Test case "sles4sap_gnome_saptune_baremetal" is ignored as there is no ipmi backend in openqawork15.
2. all-in-one means set MR_TEST="1410736,1680803,1771258,1805750,1980196,2161991,2534844,2578899,2684254,3024346,900929,941735,SAP_BOBJ,BOBJ,S4HANA-APPSERVER,MAXDB,NETWEAVER,S4HANA-DBSERVER,solutions,2382421,notes,overrides,HANA,NETWEAVER+HANA,S4HANA-APP+DB,SAP-ASE,delete_rename"
3. If you think this all-in-one test case takes too long (8 hrs) we can separate it again via settings 'MR_TEST' without code change after the merging of this PR, for example MR_TEST="1410736,1680803,...".
4. 90+% of code in new file mr_test_run.pm is moved from original mr_test.pm

- Related ticket: https://jira.suse.com/browse/TEAM-6726
- Needles: NA
- job_templates
  https://openqaworker15.qa.suse.cz/admin/job_templates/21?  (added 4 new case: 'sles4sap_gnome_saptune_mr_test_*')
- Verification run:
  https://openqaworker15.qa.suse.cz/tests/29017 (mr_test: 27 modules/cases in total, all-in-one; dropped now as it takes about 10hrs)
  http://openqaworker15.qa.suse.cz/tests/29194 (notes, 01:55 hours)
  http://openqaworker15.qa.suse.cz/tests/29195 (solutions, 01:21 hours)
  http://openqaworker15.qa.suse.cz/tests/29196 (delete_rename, 43:51 minutes)
  http://openqaworker15.qa.suse.cz/tests/29197 (overrides, 02:12 hours)

IPA added (all test modules are passed):
https://openqaworker15.qa.suse.cz/tests/33945 (added IPA parsing for 'delete_rename')
http://openqaworker15.qa.suse.cz/tests/33946 (added IPA parsing for 'notes')
https://openqaworker15.qa.suse.cz/tests/33955 (added IPA parsing for 'solutions')
http://openqaworker15.qa.suse.cz/tests/33947 (added IPA parsing for 'overrides')

IPA added (make test module failed on note 1680803 but it does not block other notes' testing):
http://openqaworker15.qa.suse.cz/tests/35606#step/test_note_1680803/1

### CODE REVIEW

File: tests/sles4sap/saptune/mr_test.pm [hunk 1]
```diff
@@ -36,7 +37,12 @@ sub setup {
     # saptune is not installed by default on SLES4SAP 12 on ppc64le and in textmode profile
     zypper_call "-n in saptune" if ((is_ppc64le() and is_sle('<15')) or check_var('DESKTOP', 'textmode'));
     # Install mr_test dependencies
-    zypper_call "-n in python3-rpm";
+    # 'zypper_call "-n in python3-rpm"' returns error message:
+    #   "There are running programs which still use files and libraries deleted or updated by recent upgrades.
+    #   They should be restarted to benefit from the latest updates.
+    #   Run 'zypper ps -s' to list these programs."
+    # Then choosing 'script_run' instead.
+    script_run "zypper -n in python3-rpm";
```
Comment (alvarocarvajald) [inline tests/sles4sap/saptune/mr_test.pm:45]: If you know the error code of `zypper` when installing `python3-rpm` you can perhaps use `zypper_call('in python3-rpm', exitcode => [0, $expected_retval])`.

**Review by alvarocarvajald — COMMENTED** [review]
Just a couple of suggestions.

Also, can you add some regression tests from some of the jobs defined in https://openqa.suse.de/group_overview/146?

**Comment by lilyeyes** [comment]
> Just a couple of suggestions.
> 
> Also, can you add some regression tests from some of the jobs defined in https://openqa.suse.de/group_overview/146?
VRs for sles15sp5 build19.1:
https://openqa.nue.suse.com/tests/9525097# (sles4sap_gnome_saptune_note_1410736, PASS)
https://openqa.nue.suse.com/tests/9525096# (sles4sap_gnome_saptune_solution_S4HANA-APP+DB, PASS)

File: lib/mr_test_run.pm [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+  get_solutions
+);
+
+sub reboot_wait {
+    my ($self) = @_;
+
+    $self->reboot;
+
+    # Wait for saptune to tune everything
+    my $timeout = 60;
```
[the remark this answers is not in the document]
Comment (lilyeyes) [reply lib/mr_test_run.pm]: Deleted.

File: tests/sles4sap/saptune/mr_test.pm (hunk 1 above)
Comment (lilyeyes) [reply tests/sles4sap/saptune/mr_test.pm]: Done. Good to know this workaround. Thanks.

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: lib/mr_test_run.pm [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+        # Skip 1805750 (SYB: Usage of HugePages on Linux Systems with Sybase ASE)
+        # The ASE docs don't recommend any specific value or formula, so it must
+        # be tested with an override file
+        $self->test_note($test) if ($test ne "1805750");
+        $self->test_override($test);
+    }
+    elsif ($test =~ m/^(x86_64|ppc64le)$/) {
+        $self->test_x86_64 if (is_ipmi);
+        $self->test_ppc64le if is_ppc64le();
+        $self->test_bsc1152598 if is_sle('>12-SP3');
```
Comment (ricardobranco777) [inline lib/mr_test_run.pm:517]: I believe we can drop the `if is_sle('>12-SP3')`

**Review by ricardobranco777 — APPROVED** [review]
One small suggestion.  Otherwise LGTM.  Great PR!

**Pull request merged**